src/audio/mixin_mixout/mixin_mixout.c

```diff
@@ -198,8 +198,10 @@ static struct comp_dev *mixout_new(const struct comp_driver *drv,
 
 	memcpy_s(&md->base_cfg, sizeof(md->base_cfg), spec, sizeof(md->base_cfg));
 
+	/* align the allocation size to a cache line for the coherent API */
 	md->mixed_data_info = rzalloc(SOF_MEM_ZONE_RUNTIME_SHARED, 0, SOF_MEM_CAPS_RAM,
```

Comment:
This is now correct. I do wonder if we should actually enforce coherent objects to be initialized with a cached object. It seems very error prone that user needs to allocate the object in a very specific way (that is really specific to coherent API and should be hidden by it). @lyakh @ranj063 @lgirdwood ?

Reply:
@kv2019i I tried that and I removed that attempt. I'm not 100% sure any more why. I think one of the reasons was that you actually want to store the uncached pointer to that memory, that's what you use. You pass the uncached pointer around, you use it to *acquire* a cached version, then you *release* it, and you never store the cached pointer. So, if you allocate a cached pointer, then you use `cache_to_uncache()` to get its uncached alias and store it. Then in the end you free it, for which you now only have your uncached pointer, so you have to convert it back to cached or otherwise `heap_free()` won't synchronise its cache... You could have `coherent_init()` allocate memory for you and add `coherent_free()` to correctly free it, but the coherent API actually only knows about its header, not the whole object... So, it's possible but not particularly elegant